lib/KBox/src/pages/NavigationPage.cpp

```diff
@@ -0,0 +1,305 @@
+/*
+  The MIT License
+
+  Copyright (c) 2016 Thomas Sarlandie thomas@sarlandie.net
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+/* AutoPilot functionality can exist at several levels.  In this initial iteration, the design 
+ focused on heading the boat on a defined course that is automatically set when AP functionality 
+ is turned on (referred to as "Heading" mode below.  Later on we could introduce an additional 
+ mode when a course to a waypoint is provided by a NMEA-connected device.  In addition, we have 
+ "Dodge" mode which suspends rudder actions temporarily in order to manually steer the boat around
+ an object or to manually put it back on course.  Waypoint functionality will be added at a later time. RIGM
+ */
+
+#include "NavigationPage.h"
+#include "MFD.h"
+#include <stdio.h>
+#include "KBoxDebug.h"
+
+
+bool apMode = false;
+bool apDodgeMode = false;
+bool apHeadingMode = false;
+bool apWaypointMode = false;//reserved for future use
+String apModeString = "APMode: Off"; // Modes: "Off", "Heading" == follow set heading, "Waypoint" = GoTo provided Waypoint, "Dodge"
+String apWaypointString = "045 20.2N, 081 50.3W";//placeholder for future use
```

Comment:
Do you mean the apWayMode boolean (which we will need later), or the apWaypointString.  I put data in the string just for kicks, but could take the string out completely or just initialize it to "".